_data-prepper/pipelines/configuration/sources/kinesis.md

```diff
@@ -0,0 +1,157 @@
+`codec` | Yes      | Codec    | The [codec](#codec) to apply.
+`buffer_timeout` | No       | Duration | The amount of time allowed for writing events to the Data Prepper buffer before timeout occurs. Any events that the source cannot write to the buffer during the specified amount of time are discarded. Default is `1s`.
+`records_to_accumulate` | No       | Integer  | The number of messages that accumulate before being written to the buffer. Default is `100`.
+`consumer_strategy` | No       | String   | Consumer strategy to use for ingesting Kinesis data streams. Default is `fan-out`. However, `polling` can also be used. if `polling` is enabled, additional configuration for `polling` will need to be added.
+`polling` | No       | String   | if `consumer_strategy` is set to `polling`, this config will need to be added. Refer to [polling](#polling).
+
+
+### Streams
+
+Use the following options in the `streams` array.
+
+Option | Required | Type | Description
+:--- |:---------| :--- | :---
+`stream_name` | Yes      | String | The name of each Kinesis stream.
+`initial_position` | No       | String | The position from where `kinesis` source starts reading stream records. `LATEST` starts reading from the most recent stream record. `EARLIEST` starts reading from the begining of the stream. Default is `LATEST`.
+`checkpoint_interval` | No       | Duration | Periodically checkpoint Kinesis streams to avoid duplication of record processing. Default is `PT2M`.
+
+## codec
+
+The `codec` determines how the `kinesis` source parses each Amazon Kinesis Record. For increased and more efficient performance, you can use [codec combinations]({{site.url}}{{site.baseurl}}/data-prepper/common-use-cases/codec-processor-combinations/) with certain processors.
+
+### `newline` codec
+
+The `newline` codec parses each single line as a single log event. This is ideal where each Kinesis stream record is processed as a single line. It also matches well when used with the [parse_json]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/parse-json/) processor to parse each line. 
+
+Use the following options to configure the `newline` codec.
+
+Option | Required | Type    | Description
+:--- | :--- |:--------| :---
+`skip_lines` | No | Integer | The number of lines to skip before creating events. You can use this configuration to skip common header rows. Default is `0`.
+`header_destination` | No | String  | A key value to assign to the header line of the stream event. If this option is specified, then each event will contain a `header_destination` field.
+
+### polling
+Option | Required | Type    | Description
+:--- | :--- |:--------| :---
+`max_polling_records` | No | Integer | The number of records to fetch from Kinesis during a single call to get Kinesis stream records.
+`idle_time_between_reads` | No | Integer  | The time duration to sleep in between calls to get Kinesis stream records. 
```

Comment:
This is a `Duration`, not an `Integer`.

Reply:
Thanks @dlvenable. I have made the changes.